physical/gcs/gcs.go

```diff
@@ -214,3 +257,271 @@ func (g *GCSBackend) List(prefix string) ([]string, error) {
 
 	return keys, nil
 }
+
+// GCSLock implements a lock using an GCS client.
+type GCSLock struct {
```

Comment:
It's up to you, but I found moving the HA bits into a separate file (gcs_ha.go) helped make the organization easier. Obviously your choice. Other backends don't do that.